docs/src/developers_guide/coding_resources/documentation_style.md

```diff
@@ -0,0 +1,146 @@
+<!-- TODO: Edit this -->
+
+> *"Omit needless words"*
```

Comment:
My opinion is that we should follow the rust documentation conventions as closely as possible.

Just to linking to the relevant RFCS, and documenting any additional requirements we have is sufficient, and most of this can be removed. I don't think the Rust guidelines are unclear enough to warrant maintaining our own complete docs style guide.